src/ast/ast.h

```diff
@@ -271,6 +273,41 @@ class NegativeInteger : public Node {
   const int64_t value;
 };
 
+class NamedParameter : public Node {
```

Comment:
I wouldn't add a new AST structure here. You are desugaring to something new that doesn't otherwise exist, which is strange.

I would desugar directly to the map or whatever the underlying implementation is, and then have the namedparameter pass emit a pass output without whatever named parameter metadata you need to consume by e.g. the resource pass.

Reply:
Done.